daemon/policy.go

```diff
@@ -216,7 +217,11 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (uint64, err
 		metrics.PolicyImportErrors.Inc()
 		logger.WithError(err).WithField("prefixes", prefixes).Warn(
 			"Failed to allocate identities for CIDRs during policy add")
-		return d.policy.GetRevision(), err
+
+		d.policy.Mutex.Lock()
```

Comment:
Why is the policy revision necessary here? We don't attempt to return a nonzero policy revision in the other error cases, because an error implies that there is no policy revision corresponding to the changes requested in this API request.

Reply:
Agreed with @joestringer , we should just return 0 here.

Reply:
Well, does anyone know why we returned `d.policy.GetRevision()` to begin with? This was just ensuring the lock was respected.

Reply:
It's a bug, let's return 0.